### DISCUSSION TITLE
Stealing 5v for Arduino from stepper controller

### DISCUSSION THREAD

**Discussion by mockendon** [kachurovskiy/nanoels#34] [Q&A] [answered]
Is this a bad idea?  The Arduino, display and encoder draw less than 50mA so I opened my stepper controller. Found a 5v line and am using it to power the other electronics (see the connection to the pin header at the opposite side of the board). Works great, but is it stupid for a reason that I don’t yet know?

![199BABB8-A0EF-4A9A-9D6E-5C4040906238](https://user-images.githubusercontent.com/12686546/146809133-94bed95d-40d2-4307-b668-94547dcb54f4.jpeg)

**Comment by kachurovskiy** [comment, marked as answer]
Matt, if it works it works :)

On the other side I had something similar in my lathe - tachometer powered from the main board. It worked until I accidentally shorted the tachometer to the lathe housing and that burned both of them. Needless to say the board was much more expensive and harder to replace. Ever since I stay clear of such optimizations :)

In your case there are the following considerations:
- Do you want ELS to shut down when motors are down? Sometimes it's more convenient to have ELS stay powered when e.g. lathe chuck guard is opened, even though it should restore all state on startup
- Driver 5V could be noisy due to all the work it does with the motors though Arduino is not too sensitive to that
- I assume you'll re-attach the driver housing because normally it's made from metal to keep the driver from adding interference to all other non-shielded wires in the vicinity

None of them are blocking though.

**Reply by mockendon** [reply]
I have set up to have nanoELS with the stepper drover powered separately from the lathe as otherwise I cannot guarantee not to loose position when the lathe is stopped.  Yes I replaced the cover but it is only plastic (DM556T Digital Stepper Driver Controller from StepperOnline).

**Comment by ws786873** [comment]
I think this is a risk. Maybe it is safer to add a step-down circuit from the 48V power supply to obtain 5V. Because I don't know what trouble it would be to bring power from the drive.

**Reply by digiexchris** [reply]
just a note: I have a closed loop stepper driver that has an encoder input. it breaks the 5v line out into a terminal, so you can power the encoder with it. I run my MCU off of that.
